determined-ai/determined
# chore: migrate agents/resource pool stores to observables

## Description

[WEB-790]

Brings the `Agents` and `ResourcePool` stores into a single `Cluster` store and migrates the state to use observables.

the store consists of the unparameterized responses for `getAgents` and `getResourcePools`, along with derived data. since there are only two APIs to call, both without params, I decided to initial polling when the provider mounts, then have the derived data as observables that `select` the appropriate transformations.

when pages need want to ensure data, they can `useRefetchClusterData()`, care should be taken to only do this at or near the top level of the component tree to avoid redundant API calls.






## Test Plan

- go to netlify preview, 
- create experiments, 
- verify that cluster allocation on sidebar is upated
- go to cluster page
- verify that the overall cluster allocation bar reflects the new experiment
- verify that the resource pool allocation cards are updated
- click into resource pool
- verify that the allocation bar at the top reflects the new experiment
- kill the experiment
- verify that the killing of the experiment is reflected in those same places





## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket






[WEB-790]: https://determinedai.atlassian.net/browse/WEB-790?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
